src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,27 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
+    // the union type field suffix is immutable
+    static size_t union_suffix_len = strlen(UnionTypeFieldSuffix());
+    // early return if no case transformation required
+    if (opts_.cpp_object_api_field_case == IDLOptions::Case_Unchanged)
+      return EscapeKeyword(field.name);
+    std::string name = field.name;
+    if (field.value.type.base_type == BASE_TYPE_UTYPE &&
+        name.length() > union_suffix_len)
+      // do not change the case style of the union type field suffix
+      name.erase(name.length() - union_suffix_len, union_suffix_len);
+    if (opts_.cpp_object_api_field_case == IDLOptions::Case_Upper)
+      name = MakeCamel(name, true); /* upper */
+    else if (opts_.cpp_object_api_field_case == IDLOptions::Case_Lower)
+      name = MakeCamel(name, false); /* lower */
+    if (field.value.type.base_type == BASE_TYPE_UTYPE)
+      // restore the union field type suffix
```

Comment:
It is hard to read `if`-code that is broken by the comment line.
Please move this comment before `if` statement or add `{}` scope block.

CSG (https://google.github.io/styleguide/cppguide.html#Conditionals):
```cpp
// Bad - IF statement is too long to omit braces
if (condition)
  // Comment
  DoSomething();
```

Reply:
resolved in next commit